### PR TITLE
Refactor EinsumLayerTest, EyeLayerTest

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Refactor EinsumLayerTest, EyeLayerTest
 - Update gpu tests instantiations

### Tickets:
 - [CVS-111364](https://jira.devtools.intel.com/browse/CVS-111364)

### CODE REVIEW

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/einsum.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        std::vector<InputShape>   // Input shapes
+> EinsumEquationWithInput;
+
+typedef std::tuple<
+        ov::element::Type,         // Model type
+        EinsumEquationWithInput,   // Equation with corresponding input shapes
+        std::string                // Device name
+> EinsumLayerTestParamsSet;
+
+class EinsumLayerTest : public testing::WithParamInterface<EinsumLayerTestParamsSet>,
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/einsum.hpp:26]: Minor: Please align

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/eye.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+using EyeLayerTestParams = std::tuple<
+    std::vector<ov::Shape>,  // eye shape
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/eye.hpp:11]: Use `vector::InputShape` instead to support dynamic case

**Review by iefode — CHANGES_REQUESTED** [review]

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/eye.hpp (hunk 2 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/eye.hpp:11]: This test model doesn't have parameter nodes at all. It uses constants as inputs. So we don't need dynamic cases here

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/einsum.hpp (hunk 1 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/einsum.hpp:26]: done

**Review by vurusovs — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Pull request merged**